sys/linux/io_uring.txt

```diff
@@ -96,6 +133,41 @@ io_uring_files_update {
 	fds	ptr64[in, array[fd]]
 }
 
+#
+# type template for io_uring_restriction
+#
+
+type io_uring_restriction[OPCODE, OPARG] {
+	op	const[OPCODE, int16]
+	oparg	OPARG
+	resv	const[0, int8]
+	resv2	array[const[0, int32], 3]
+}
+
+io_uring_restriction_st [
+	ioring_restriction_register_op		io_uring_restriction[IORING_RESTRICTION_REGISTER_OP, flags[io_uring_register_opcodes, int8]]
+	ioring_restriction_sqe_op		io_uring_restriction[IORING_RESTRICTION_SQE_OP, flags[io_uring_register_opcodes, int8]]
+	ioring_restriction_sqe_flags_allowed	io_uring_restriction[IORING_RESTRICTION_SQE_FLAGS_ALLOWED, flags[io_uring_register_opcodes, int8]]
+	ioring_restriction_sqe_flags_required	io_uring_restriction[IORING_RESTRICTION_SQE_FLAGS_REQUIRED, flags[io_uring_register_opcodes, int8]]
+]
+
+io_uring_rsrc_register {
+	nr	len[data, int32]
+	flags	int32
+	resv2	int64
+	data	ptr[in, array[iovec_out]]
```

Comment:
s/ptr/ptr64/

Reply:
Gotcha, I was too careless to pay attention to the `__aligned_u64` type.
Will fix it, thanks